### PR TITLE
Provide an option to fully mask the IP address

### LINKED ISSUE #16377: Offer possibility to not store IP address at all
It's already possible to anonymise 3 bytes: 
image

This is however not good enough. We also need to have a setting to basically anonymise 4 bytes and therefore the entire IP address. In this case we'd for example always track `0.0.0.0`. 

Why might 3 bytes not be enough? Eg if you have only few visitors at a certain time, and you have an IP address in Matomo logged starting with eg `192.0.0.0`, then you might still be able to crosslink this with log files to get the full IP address etc.

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #16377

### CODE REVIEW

File: plugins/PrivacyManager/IPAnonymizer.php [hunk 1]
```diff
@@ -25,6 +26,11 @@ class IPAnonymizer
      */
     public static function applyIPMask(IP $ip, $maskLength)
     {
+        // IPv6 can't handle anonymizing 4 bytes
+        if ($maskLength == 4 && $ip instanceof IPv6 && !$ip->isMappedIPv4()) {
```
Comment (tsteur) [inline plugins/PrivacyManager/IPAnonymizer.php:30]: @sgiehl wouldn't we add this maybe to the IPv6 class? (I suppose IPv4 already supports it)

Comment (sgiehl) [reply plugins/PrivacyManager/IPAnonymizer.php:30]: created https://github.com/matomo-org/component-network/pull/15
We will need to publish a new release and update the component here after merging...

Comment (tsteur) [reply plugins/PrivacyManager/IPAnonymizer.php:30]: 👍  left a comment in that PR. Feel free to merge the component PR afterwards, create the release and then update this PR

Comment (sgiehl) [reply plugins/PrivacyManager/IPAnonymizer.php]: updated

**Comment by tsteur** [comment]
works

**Pull request merged**